backend/app/controllers/products_controller.rb

```diff
@@ -0,0 +1,65 @@
+class ProductsController < ApplicationController
+    include Dry::Monads::Result::Mixin
+
+    #POST /products
+    def create
+        case product_creation
+        in Success(product)
+          render json: prepare_json(product), status: :created
+        in Failure(error)
+          render json: { errors: error.messages }, status: :unprocessable_entity
+        end
+    end
+
+    #PUT/PATCH /products/edit/1
+    def update
+        case set_product
+        in Success(product)
+            render json: prepare_json(product), status :updated
+        in Failure
+            render json: { error: error.messages }, status :unprocessable_entity
+        end
+    end
+
+    #GET /products/1
+    def show
+        id = params.extract_value(:id)
+        @product = product.find(id)
```

Comment:
What will happen if there is no record with requested id? And what do you gonna do with that?